### PR TITLE
Immutable splice

### PR DESCRIPTION
**Pull request by evnp**
Adds some array manipulation utilities

### CODE REVIEW

File: test/util.js [hunk 1]
```diff
@@ -19,3 +26,97 @@ describe('pluralize()', function() {
     expect(pluralize('die', 4, 'dice')).to.equal('dice');
   });
 });
+
+describe('immutableSplice()', function() {
```
Comment (tdumitrescu) [inline test/util.js]: awesome, love all the testing. Can we have a couple example-based cases for `immutableSplice` to make it super-simple to look at the tests and see how it's supposed to look in practice? something like:
```javascript
it('removes items', function() {
  expect(immutableSplice([1,2,3,4,5], 2, 2)).to.eql([1,2,5]);
});
```
and similar for inserting

Comment (evnp) [inline test/util.js]: Good call ce58f9e

**Comment by evnp** [comment]
No worries, should be pretty easy s///g

**Comment by tdumitrescu** [comment]
published 0.6.4, merci

**Pull request merged**